paasta_tools/api/tweens/request_logger.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Creates a tween that logs information about requests.
+"""
+import json
+import logging
+import time
+import traceback
+
+import pyramid
+
+from paasta_tools import utils
+from paasta_tools.api import settings
+
+try:
+    import clog
+except ImportError:
+    clog = None
+
+
+DEFAULT_REQUEST_LOG_NAME = "tmp_paasta_requests"
+
+
+def includeme(config):
+    if clog is not None:
+        config.add_tween(
+            "paasta_tools.api.tweens.request_logger.request_logger_tween_factory",
+            under=pyramid.tweens.INGRESS,
+        )
+
+
+class request_logger_tween_factory:
+    """Tween that logs information about requests"""
+
+    def __init__(self, handler, registry):
+        self.handler = handler
+        self.registry = registry
+        self.log_name = registry.settings.get(
+            "request_log_name", DEFAULT_REQUEST_LOG_NAME,
+        )
+
+    def _log(
+        self, endpoint, level=logging.INFO, additional_fields=None,
+    ):
+        if clog is not None:
+            dct = {
+                "timestamp": time.time(),
+                "hostname": utils.get_hostname(),
+                "level": logging.getLevelName(level),
```

Comment:
let's not do potentially expensive calculations every time we log a request. it's not like hostname or loglevel is going to chnage much during server runtime.

Reply:
I moved the `hostname` to api settings, so that it gets set once when the API starts up. I also removed the extra level processing, and now the level is passed in as a string directly.